src/Dfe.PlanTech.Application/Cookie/Service/CookiesCleaner.cs

```diff
@@ -0,0 +1,26 @@
+using Dfe.PlanTech.Domain.Cookie;
+using Microsoft.AspNetCore.Http;
+
+namespace Dfe.PlanTech.Application.Cookie.Service;
+
+///<remarks>
+///Removes non-essential cookies from the browser when user rejects cookies.
+///</remarks>
+public class CookiesCleaner(CookiesCleanerOptions opts) : ICookiesCleaner
+{
+    public void RemoveNonEssentialCookies(HttpContext context)
+    {
+        var nonessentialCookies = GetNonEssentialCookies(context);
+
+        foreach (var nonessentialCookie in nonessentialCookies)
+        {
+            context.Response.Cookies.Append(nonessentialCookie, "", CreateExpiryCookieOptions());
+        }
+    }
+
+    private static CookieOptions CreateExpiryCookieOptions()
+      => new() { Expires = DateTime.Now.AddSeconds(1) };
```

Comment:
what about the secure and httponly flags - security hotspots from SonarCloud

Reply:
These don't matter, as we're not exactly creating a cookie per se, we're just setting the value to be empty, and giving it a very short lived lifespan, so that it overrides the existing one and then gets deleted.

However, I've amended it to stop Sonarcloud complaining. Also added comments for clarity.